packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -115,10 +115,128 @@ const deferActions = actions => {
   return deferred
 }
 
-const getLocalReporter = (activity, reporter) =>
-  activity
-    ? { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
-    : reporter
+/**
+ * Create a local reporter
+ * Used to override reporter methods with activity methods
+ */
+function getLocalReporter({ activity, reporter }) {
+  // If we have an activity, bind panicOnBuild to the activities method to
+  // join them
+  if (activity) {
+    return { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
+  }
+
+  return reporter
+}
+
+function extendErrorIdWithPluginName(pluginName, errorMeta) {
+  if (typeof errorMeta === `object`) {
+    const id = errorMeta && errorMeta[`id`]
+
+    if (id) {
+      errorMeta[`id`] = `${pluginName}_${id}`
+    }
+  }
+}
+
+function getErrorMapWthPluginName(pluginName, errorMap) {
+  const entries = Object.entries(errorMap)
+
+  return entries.reduce((memo, [key, val]) => {
+    memo[`${pluginName}_${key}`] = val
+
+    return memo
+  }, {})
+}
+
+function extendLocalReporterToCatchPluginErrors({
+  reporter,
+  pluginName,
+  runningActivities,
+}) {
+  let setErrorMap
+
+  let error = reporter.error
+  let panic = reporter.panic
+  let panicOnBuild = reporter.panicOnBuild
+
+  if (pluginName && reporter?.setErrorMap) {
+    setErrorMap = errorMap => {
+      const newErrorMap = getErrorMapWthPluginName(pluginName, errorMap)
```

Comment:
Nit: when I would read this I would assume it's a clone of the given error map. But it's not since the structure of the key names is different. (In other words; if given its own output and just reading this call, I would expect this function to return a cloned object with the same shape)